### PR TITLE
Fix service url builder unnecessarily encoding query param string

### PR DESCRIPTION
**Pull request by emswbandara**
This PR fixes the issue where DefaultServiceURLBuilder unncessarily encoding the query param string of a URL. As a result URL is getting as :
https://localhost:9443/samlsso?sessionDataKey%3Dac584d70-8453-426c-945c-8d8b097387da

Expected one should be  :
https://localhost:9443/samlsso?sessionDataKey=ac584d70-8453-426c-945c-8d8b097387da

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java [hunk 1]
```diff
@@ -501,13 +501,7 @@ private void appendParamsToUri(StringBuilder serverUrl, String resolvedParamsStr
...
             }
             if (StringUtils.isNotBlank(resolvedParamsString)) {
-                try {
-                    serverUrl.append(delimiter).append(URLEncoder.encode(resolvedParamsString,
-                            StandardCharsets.UTF_8.name()));
-                } catch (UnsupportedEncodingException e) {
-                    throw new URLBuilderException(String.format("Error while trying to build url. %s is not supported" +
-                            ".", StandardCharsets.UTF_8.name()), e);
-                }
+                serverUrl.append(delimiter).append(resolvedParamsString);
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java:504]: Do we encode the parameters when building resolvedParamsString?

Comment (emswbandara) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java:504]: do you mean individual parameters?

Comment (mefarazath) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java:504]: Yes

Comment (emswbandara) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java:510]: It was not encoded previously. Added fix at https://github.com/wso2/carbon-identity-framework/pull/3011/commits/f4f1ea2d22cbde21781861b96a7a230a312daada

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java [hunk 2]
```diff
@@ -260,33 +260,32 @@ public void testBuild() {
 
         return new Object[][]{
                 {"https", "www.wso2.com", 9443, "/proxyContext", "abc", false, null, "", fragmentParams,
-                        "https://www.wso2.com:9443/proxyContext#key1%3Dfragment%26key2%3Dfragment%26key3%3Dfragment" +
-                                "%26key4%3Dfragment", ""},
+                        "https://www.wso2.com:9443/proxyContext#key1=fragment&key2=fragment&key3=fragment&key4" +
```
Comment (mefarazath) [inline components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilderTest.java:263]: Can we add a test case where a parameter has a special character like "space" or some character that needs encoding?

**Pull request merged**